pkg/cli/destroy/destroy.go

```diff
@@ -240,38 +258,127 @@ func Destroy(ctx context.Context, opts DeleteOptions) error {
-		list, err := resourceClient.List(ctx, metav1.ListOptions{})
-		if err != nil {
-			return false, err
-		}
-		if len(list.Items) == 0 {
-			return true, nil
+	//  function to poll with wait.ExponentialBackoffWithContext
+	deleteAndWait := func(deleteFinalizers bool) func(ctx context.Context) (bool, error) {
+		// log each key as waiting only once on the info level, and continue logging on the debug level
+		loggedDeletion := sets.New[string]()
+		infofOnceThenDebugf := func(str string, args ...interface{}) {
+			logLine := fmt.Sprintf(str, args...)
+			if loggedDeletion.Has(logLine) {
+				log.Debug(logLine)
+				return
+			}
+			log.Info(logLine)
+			loggedDeletion.Insert(logLine)
 		}
-		for _, object := range list.Items {
-			if !object.GetDeletionTimestamp().IsZero() {
-				return false, nil
+
+		return func(ctx context.Context) (bool, error) {
+			infofOnceThenDebugf("checking all %q", resource)
+
+			// fetch all
+			resourceClient := dynamicClient.Resource(gvr)
+			list, err := resourceClient.List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return false, err
 			}
-			if object.GetNamespace() == "" {
-				log.Infof("deleting %v: %v", resource, object.GetName())
-			} else {
-				log.Infof("deleting %v: %v/%v", resource, object.GetNamespace(), object.GetName())
+			// succeed when all resources are deleted
+			if len(list.Items) == 0 {
+				return true, nil
 			}
-			err := resourceClient.Namespace(object.GetNamespace()).Delete(ctx, object.GetName(), metav1.DeleteOptions{})
-			if err != nil && !kerrors.IsNotFound(err) {
-				return false, err
+
+			isVCluster := resource == "virtualclusterinstances"
+
+			// delete all resources and log deleting resources
+			for _, object := range list.Items {
+				// get namespaced name
+				namespacedName := object.GetName()
+				namespace := object.GetNamespace()
+				if namespace != "" {
+					namespacedName += "/" + namespace
+				}
+				isExternalVCluster := false
+				virtualClusterInstance := &storagev1.VirtualClusterInstance{}
+				if isVCluster {
+					//convert unstructured to VirtualClusterInstance
+					err = runtime.DefaultUnstructuredConverter.FromUnstructured(object.Object, &virtualClusterInstance)
+					if err != nil {
+						log.Warnf("couldn't cast %q object %q to VirtualClusterInstance: %v", resource, namespacedName, err)
+					}
+					isExternalVCluster = virtualClusterInstance.Spec.External
+				}
+				if isExternalVCluster && virtualClusterInstance.Status.VirtualCluster != nil {
+					vConfig := &config.Config{}
+					err = config.UnmarshalYAMLStrict([]byte(virtualClusterInstance.Status.VirtualCluster.HelmRelease.Values), vConfig)
+					if err != nil {
+						return false, fmt.Errorf("failed to unmarshal virtual cluster config for %v %q: %w", resource, namespacedName, err)
+					}
+					if !nonInteractive {
+						if vConfig.ControlPlane.BackingStore.Database.External.Enabled && vConfig.ControlPlane.BackingStore.Database.External.Connector != "" {
```

Comment:
Sorry one mroe thing, you should still check this and report it if its non-interactive. Just without the "Do you want to continue?"

Reply:
Also, it look like if its not non interactive you and not using external connector, you don't show any warning. I think you want something like this:
```
if vConfig.ControlPlane.BackingStore.Database.External.Enabled && vConfig.ControlPlane.BackingStore.Database.External.Connector != "" {
	if !nonInteractive {
		yesOpt := "yes"
		noOpt := "no"
		out, err := log.Question(&survey.QuestionOptions{
			Options:  []string{yesOpt, noOpt},
			Question: fmt.Sprintf("IMPORTANT! You are removing an externally deployed virtual cluster %q from the platform.\n It will not be destroyed as the deployment is managed externally, but its connection to its database will be removed.\nDo you want to continue?", namespacedName),
		})
		if err != nil {
			return false, fmt.Errorf("failed to prompt for confirmation: %w", err)
		}
		if out != yesOpt {
			return false, fmt.Errorf("destroy cancelled during prompt")
		}
	}
} else {
	log.Warnf("removing an externally deployed virtual cluster %q from the platform. It will not be destroyed as the deployment is managed externally, but its connection to its database will be removed.", namespacedName)
}
```

Reply:
Fixed in latest commit